File: 2085.统计出现过一次的公共字符串.py
#
# @lc app=leetcode.cn id=2085 lang=python3
#
# [2085] 统计出现过一次的公共字符串
#

# @lc code=start
class Solution:
    def countWords(self, words1: List[str], words2: List[str]) -> int:
        res = 0
        words1_dict = {}
        words2_dict = {}
        for one in words1:
            if one in words1_dict:
                words1_dict[one] += 1
            else:
                words1_dict[one] = 1
        for one in words2:
            if one in words2_dict:
                words2_dict[one] += 1
            else:
                words2_dict[one] = 1
        for key, val in words1_dict.items():
            if val == 1 and key in words2_dict and words2_dict[key] == 1:
                res += 1
        return res
# @lc code=end

File: 2085.统计出现过一次的公共字符串.go
/*
 * @lc app=leetcode.cn id=2085 lang=golang
 *
 * [2085] 统计出现过一次的公共字符串
 */

// @lc code=start
func countWords(words1 []string, words2 []string) (res int) {
	words1Map := make(map[string]int)
	words2Map := make(map[string]int)
	for _, one := range words1 {
		words1Map[one]++
	}
	for _, one := range words2 {
		words2Map[one]++
	}
	for key, val := range words1Map {
		if val == 1 {
			if v, ok := words2Map[key]; ok {
				if v == 1 {
					res++
				}
			}
		}
	}
	return res
}

// @lc code=end

